saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/search/searchFilters.js

```diff
@@ -245,9 +258,30 @@ var SearchFilters = (function(){
 	},
 	
 	/* Clear target field e.g. input select */
-	clearField = function($field) {
-		
-		if ($field.is("input:checkbox")) {
+	clearField = function($field, filterValue) {
+
+		if ($field.is("input:hidden")) {
+			var value = $field.attr("value"),
+				array = value.split(", "),
+				length = array.length,
+				newValue = "",
+				first = true;
+			
+			for (var i = 0; i < length; i += 1) {
+				if (filterValue != array[i]) {
+					
+					if (!first) {
+						newValue += ", ";
+					}
+					
+					newValue += array[i];
+					first = false;
+				}
+			}
+			
+			$field.attr("value", newValue);
```

Comment:
Wouldn't it be clearer to use a string replace function? Something like value=value.replace(filterValue+", ", "")?